sdv/constraints/tabular.py

```diff
@@ -379,3 +381,73 @@ def reverse_transform(self, table_data):
         table_data[self._column] = self._formula(table_data)
 
         return table_data
+
+
+class Rounding(Constraint):
+    """Round a column based on the specified number of digits.
+
+    Args:
+        column (list[str]):
+            Name of the column(s) to round.
+        digits (int or dict[str->int]):
+            How much to round each column. If an `int` is provided, all columns
+            will be rounded to that number of digits. If a `dict` that maps column
+            to digits is provided, each column will be represented to the specified
+            number of digits.
+        handling_strategy (str):
+            How this Constraint should be handled, which can be ``transform``
+            or ``reject_sampling``. Defaults to ``transform``.
+        tolerance (int):
+            How many differences in decimal places we will tolerante when
+            reject sampling.
+    """
+
+    def __init__(self, columns, digits, handling_strategy='transform', tolerance=1):
+        self._columns = columns
+        self._digits = digits
+        self._tolerance = tolerance
+        super().__init__(handling_strategy=handling_strategy, fit_columns_model=False)
+
+    def is_valid(self, table_data):
+        """Determine if the data satisfies the rounding constranit.
+
+        Args:
+            table_data (pandas.DataFrame):
+                Table data.
+
+        Returns:
+            pandas.Series:
+                Whether each row is valid.
+        """
+        table_data = table_data.copy()
+        d_places = table_data[self._columns].applymap(
+            lambda d: -1 * decimal.Decimal(str(d)).as_tuple().exponent)
+
+        cols = [self._columns] if isinstance(self._digits, int) else [col for col in self._columns]
```

Comment:
I do not understand this. It seems like the `[col for col in self._columns]` part is redundant. In any case, I would avoid doing this here: let's do it inside the `__init__` instead, and always store `self._columns` as a list from the begining.

Reply:
That makes sense. I was trying to account for both the case where `_digits` is an `int` and the case where `_digits` is a `dict`. That way if `_digits` is an `int` and applies to all the columns, we can just filter it once instead of iterating through each column and applying the filter separately.

I think the approach of making a dict in `__init__` is more readable though.